service/policy/kasregistry/key_access_server_registry.proto

```diff
@@ -351,6 +351,96 @@ message ListKeyAccessServerGrantsResponse {
   policy.PageResponse pagination = 10;
 }
 
+/*
+   KAS Key Management Requests and Response Messages
+*/
+message CreateKeyRequest {
+  // Required
+  string kas_id = 1 [(buf.validate.field).string.uuid = true];
+  string key_id = 2 [(buf.validate.field).string.min_len = 1];
+  Algorithm key_algorithm = 3 [(buf.validate.field).enum.defined_only = true];
+  KeyMode key_mode = 4 [(buf.validate.field).enum.defined_only = true];
+
+  // Optional
+  bytes public_key_ctx = 5;
+  // Optional
+  bytes private_key_ctx = 6; // Specific structure based on key provider implementation
+  // Optional
+  string provider_config_id = 7;
+
+  // Common metadata
+  common.MetadataMutable metadata = 100;
+}
+
+message CreateKeyResponse {
+  AsymmetricKey key = 1;
+}
+
+message GetKeyRequest {
+  oneof identifier {
+    option (buf.validate.oneof).required = true;
+    string id = 2 [(buf.validate.field).string.uuid = true];
+    string key_id = 3 [(buf.validate.field).string.min_len = 1];
+  }
+}
+
+message GetKeyResponse {
+  AsymmetricKey key = 1;
+}
+
+message ListKeysRequest {
+  Algorithm key_algorithm = 1 [(buf.validate.field).enum.defined_only = true];
+
+  oneof kas_filter {
+    string kas_id = 2 [(buf.validate.field).string.uuid = true];
+    string kas_name = 3;
+    string kas_uri = 4;
+  }
+
+  // Optional
+  policy.PageRequest pagination = 10;
+}
+
+message ListKeysResponse {
+  repeated AsymmetricKey keys = 1;
+
+  policy.PageResponse pagination = 10;
+}
+
+message UpdateKeyRequest {
+  // Required
+  string id = 1 [(buf.validate.field).string.uuid = true];
+  // Optional
+  KeyStatus key_status = 3 [(buf.validate.field).enum.defined_only = true];
+
+  // Optional
+  // Common metadata
+  common.MetadataMutable metadata = 100;
+  common.MetadataUpdateEnum metadata_update_behavior = 101;
+}
+
+message UpdateKeyResponse {
+  AsymmetricKey key = 1;
+}
+
+message RotateKeyRequest {
+  // Required (Current Active Key ID)
+  string active_key_id = 1 [(buf.validate.field).string.uuid = true];
```

Comment:
Is this the `kid` field or the row `id`?

Reply:
the kid. I feel like this is going to get confusing. Do you think `active_kid` is better?

Reply:
Yes, that's clearer. I also think we should capture in the proto comments that the `AsymmetricKey` fields that `id` is a uuid and `key_id` is a name something like `my-kas-key-123` that could be a uuid but is admin-defined.